iothub/service/src/Common/Exceptions/ErrorCode.cs

```diff
@@ -36,7 +36,21 @@ public enum ErrorCode
         DeviceInvalidResultCount = 400002,
 
         InvalidOperation = 400003,
+
+        /// <summary>
+        /// Returned by the service if a given method invocation request has an invalid argument in it.
```

Comment:
I'm not sure this is the exclusive reason a client will see this, and even if it is right now, it may not be in the future. I suggest making it generic, and then noting a case why it might be hit. Most importantly, we should be telling people what to do if they see it, like in your para, but I'd like to start generic and then mention specific cases so it doesn't sound like it'll only happen in the specific case.